### ISSUE TITLE
[Bug] - Some PNGs work properly, but many display "Error Reading File" / Appears to be an issue w/ text added to the PNG by extension "Dynamic Thresholding"

### ISSUE THREAD

**Issue by CCpt5** [stassius/MetaParser#1]
**Describe the bug**
Hi, first I just wanted to say that I've been looking for a tool like this for a while - thank you so much for your time and effort!!

I get "Error Reading File" for a lot of PNGs that I've generated w/ A1111. I'm on the most recent commit, but do use a few extensions that might add extra metadata that breaks functionality (?). Here are example of 3 images that Metaparser reads properly, and 3 that display "Error Reading File" from a batch of images I generated today. 

(Updated: After comparing prompts further, it appears the "Dynamic Thresholding" extension adds text that breaks Meaparser's ability to read PNGs properly):

Don't work / Display "Error Reading File"
---
https://drive.google.com/file/d/1-sU4nO5hNb3klRQ9zgC1GBCBKCdW8Av7/view?usp=sharing
https://drive.google.com/file/d/1IOJa_BxAbOZSxsG3IqkqagdBAmnSfuNR/view?usp=sharing https://drive.google.com/file/d/1g5Ldzdq5Ig4pnq1PsnDnVbBlsw9eXv73/view?usp=sharing

Work/Don't produce "Error Reading File"
---
https://drive.google.com/file/d/159vvFkL3bAoLk6dkozfdBGiGwctGFLkv/view?usp=sharing 
https://drive.google.com/file/d/1E-vyEJWo60yLn6wPEyhaftlbuoMIoORY/view?usp=sharing
https://drive.google.com/file/d/1m27Rf1uutBtWSaJ5CIK8LtcRSX6toYcr/view?usp=sharing

**To Reproduce**
1. Download PNGs from the GDrive links above
2. Drag/Drop into A1111's "PNG Info" tab and notice all PNG info is displayed w/o error
3. Drag/Drop same PNG into Metaparser and notice "Error Reading File" message
4. Repeat w/ PNGs uploaded to GDrive that work in Metaparser and compare metadate between working & not working. Appears the line added for settings related to "Dynamic Thresholding" extension break Metaparser's functionality.  The "Dynamic Thresholding" extension can be downloaded off the A1111 list the extension or this is the link to the repo:  https://github.com/mcmonkeyprojects/sd-dynamic-thresholding
5. The line "Dynamic Thresholding" adds to the PNG metadata is bolded here (from one of the included PNGs):

"Lost in the stillness, I search for a sign of life in the silence that surrounds me(" hyperrealism:0.76), cinematic lighting, volumetric light, detailed textures, 4k, unreal engine 5, cinematic, topnotch artstyle, [ IBM PC|Microsoft Windows 95|redshift|hyper-realism render] cgsociety mucha, alphonse and rutkowski greg and artgerm by art illustration, focus, sharp smooth, art
Steps: 52, Sampler: Euler a, CFG scale: 20, Seed: 2135717810, Size: 512x640, Model hash: 46265dcd64, Model: 2023-04-14 - Topnotch EyeDocDay - 42img - 5000 steps, Denoising strength: 0.15, Hires upscale: 1.4, Hires upscaler: 4x_foolhardy_Remacri, **Dynamic thresholding enabled: True, Mimic scale: 6, Threshold percentile: 99.05, Mimic mode: Cosine Up, Mimic scale minimum: 3.5, CFG mode: Cosine Up, CFG scale minimum: 3.5**
Template: __topnotch-gpt3.5__, [ __1980s-technology__|__1990s-technology__|__render-engine__|__render__]

**Expected behavior**
PNGs that open properly in A1111's "PNG Info" should open properly in Metparser

**Screenshots**
See attached PNGs

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome

**Additional context**
Related to this as an "Enhancement Request" would be adding support to for "Template " data that can be added to PNG metadata by the popular extension "Dynamic Prompts": https://github.com/adieyal/sd-dynamic-prompts

The "Template" info is an exact copy of the prompt before wildcards/dynamic prompts are selected and replaced in the prompt. There is an option in A1111's settings for that extension called: "Save template to metadata: Write prompt template into the PNG metadata". It's very helpful to be able to reproduce the exact original prompt (including wildcards used not just the one that was selected randomly that time). All of the images I linked above were generated with random wildcards. 

Of note, the metadata that "Dynamic Prompts" adds to PNGs does not seem to break Metaparser's ability to read the PNG as the metadata added for "Dynamic Thresholding" does. 

Thanks if you can look into this issue and perhaps resolve it in the future!  I only just heard about your repo, but it will be extremely helpful if it can handle all of the images I generate w/ A1111 regardless of extensions. I was brought to your repo after reading a discussion on reddit today: https://old.reddit.com/r/StableDiffusion/comments/12md6kc/is_there_a_simple_apprepoetc_that_will_add_a/

Thanks again for you time and effort on this tool!!

**Comment by stassius** [comment]
Hello! Thank you for the report. It looks like the Dynamic prompt extension doesn't use the common A1111 convention for the metadata. At the moment the convention is simple: split key-value pairs by comma and split key and value by colon. That's how Metaparser parses it.

Take a look at this string:
`CFG scale minimum: 3.5 Template: __topnotch-gpt3.5__, [ __1980s-technology__|__1990s-technology__|__render-engine__|__render__] `
Up until CFG scale minimum everything was OK. But "Template" doesn't have a comma before it, not to mention comma inside the value. 

I can make a special case for this extension and search for "Template:" and assume that the remainder is its value, but it doesn't sound like a great idea to me. I'll look into the extension source code and try to find a solution.

**Comment by stassius** [comment]
I opened an issue in the extension repo. Meanwhile I made a quick and dirty patch to fix the issue with non-standard metadata. It could show you wrong fields at the end of the table, but at least it will parse everything before them. Can't do anything else before the author of the extension fixed it on his side.
I will mark this issue as closed. If you have any additional questions or proposals, feel free to contact me.
